src/components/SuggestToWeb.js

```diff
@@ -0,0 +1,221 @@
+      comparisonTime(result.data.characters7Link, characterLink);
+    });
+  };
+
+  function comparisonTime(arrays, compareArray) {
+    // //console.log(arrays);
+    // //console.log(compareArray);
+
+    let combineCharacterInfo = [];
+
+    // Loop over the elements in `arrays`
+    for (const array of arrays) {
+      // Create a copy of the `link_skill` array
+      let linkSkills = array.link_skill.slice();
+      // Use the `unshift()` method to add the `id` at the beginning of the `link_skill` array
+      linkSkills.unshift(array.rarity)
+      linkSkills.unshift(array.type)
+      linkSkills.unshift(array.art);
+      linkSkills.unshift(array.thumb);
+      linkSkills.unshift(array.id);
+
+      // Use the `push()` method to add the modified `link_skill` array to `combineCharacterInfo`
+      combineCharacterInfo.push(linkSkills);
+    }
+    // Print `combineCharacterInfo`
+    // //console.log(combineCharacterInfo);
+
+    // makes a variable matchers that maps over the new array of JUST linkskills, then iterates over the comparison array to see if there are any matching elemets
+    let matches = combineCharacterInfo.map((otherArray) => {
+      let count = 0;
+      for (const element of otherArray) {
+        if (compareArray.includes(element)) {
+          count++;
+        }
+      }
+      return count;
+    });
+
+    //TODO: This is where we can get the if statement to pass into suggestions for links into separate divs. The id of the character is the first in the array, allowing us to pass the id through the props
+    let arrayOfAll7Matches = [];
+    let arrayOfAll6Matches = [];
+    let arrayOfAll5Matches = [];
+    let arrayOfAll4Matches = [];
+    let arrayOfAll3Matches = [];
+    matches.forEach((match, index) => {
+      if (match === 7) {
+        // //console.log(`This array: ${combineCharacterInfo[index]} matches this many times: ${match}`);
+        arrayOfAll7Matches.push(combineCharacterInfo[index]);
+      } else if (match === 6) {
+        // //console.log(`This array: ${combineCharacterInfo[index]} matches this many times: ${match}`);
+        arrayOfAll6Matches.push(combineCharacterInfo[index]);
+      } else if (match === 5) {
+        // //console.log(`This array: ${combineCharacterInfo[index]} matches this many times: ${match}`);
+        arrayOfAll5Matches.push(combineCharacterInfo[index]);
+      } else if (match === 4) {
+        // //console.log(`This array: ${combineCharacterInfo[index]} matches this many times: ${match}`);
+        arrayOfAll4Matches.push(combineCharacterInfo[index]);
+      } else if (match === 3) {
+        // //console.log(`This array: ${combineCharacterInfo[index]} matches this many times: ${match}`);
+        arrayOfAll3Matches.push(combineCharacterInfo[index]);
+      } else {
+        // //console.log(`This array: ${combineCharacterInfo[index]} matches this many times: ${match} :(`)
+      }
+    });
+    setCharactersWith7Matches(arrayOfAll7Matches);
```

Comment:
The use of arrays to package data in this function is already a bit confusing, but I'd definitely recommend avoiding 2D arrays to package data in react state. I'd recommend the following changes:

* Even though they're created from complicated logic, these arrays are still derived from the GQL query and so don't need to be react state at all! First step is to convert this into a helper function outside the component and have it return the data directly. Maybe package it in an object for access so you can do the following (rolling in some other recommended changes):
```
function SuggestToWeb({ character }) {
  const { loading: loading7Links, data: { characters7Links } } = useQuery(QUERY_7LINKS);
  const [selectedCharacterLinks, setSelectedCharacterLinks] = useState(character.link_skill);

  const { match3, match4, match5, match6, match7 } = computeCharacterMatches(characters7Links, selectedCharacterLinks);
}
```
* If I'm not mistaken, the compareArray argument should just be the currently selected character's `link_skills`, right? In that case, the dropping other character attributes into the `combineCharacterInfo` array items and using them to check if the skills exist in the other characters' link_skills is both confusing and adding unnecessary work when counting matches. I think a better way of managing the data is to skip the creation of `combineCharacterInfo` and just use `character.link_skills` when computing matches:
```
const matches = characters7Links.map(({ link_skills }) => (
  _.countBy(link_skills, selectedCharacterLinks.includes)
);
```
* Once you know how to compute matches, you can simplify further by using lodash [groupBy](https://lodash.com/docs/4.17.15#groupBy) to get the characters-per-matchcount. Then you don't need to manually package in an object, can just return the generated grouping

So to bring it all together, I think this whole function can be rewritten as
```
// creates an object of {[matchCount]: character}
function computeCharacterMatches(characters7Links, selectedCharacterLinks) {
  return characters7Links.groupBy(({ link_skills }) => (
    _.countBy(link_skills, selectedCharacterLinks.includes)
  );
```
and call it like
```
const {
  "7": match7, 
  "6": match6, 
  "5": match5, 
  "4": match4,
  "3": match3, 
} = computeCharacterMatches(characters7Links, selectedCharacterLinks);
```